enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/InternalAPIKeyAuthenticator.java

```diff
@@ -69,8 +74,20 @@ public InternalAPIKeyAuthenticator(String securityParam) {
 
     @Override
     public boolean canAuthenticate(RequestContext requestContext) {
+        String apiType = requestContext.getMatchedAPI().getApiType();
         String internalKey = requestContext.getHeaders().get(
                 ConfigHolder.getInstance().getConfig().getAuthHeader().getTestConsoleHeaderName().toLowerCase());
+        if (apiType.equalsIgnoreCase("WS") && internalKey == null) {
+            internalKey = extractInternalKeyInWSProtocolHeader(requestContext);
+
+            if (isAdditionalResponseHeadersRequired(requestContext)) {
+                HashMap<String, String> responseHeadersToAddMap = new HashMap<>();
+                responseHeadersToAddMap.put(
+                        HttpConstants.WEBSOCKET_PROTOCOL_HEADER, Constants.WS_API_KEY_IDENTIFIER);
+                requestContext.setResponseHeadersToAddMap(responseHeadersToAddMap);
```

Comment:
Here let's `getResponseHeadersToAddMap` and put values into that map. Otherwise, if some values are added to that map from another filter, we will lose them.